tests/ui/parser/issue-103869.stderr

```diff
@@ -1,16 +1,12 @@
 error: expected one of `(`, `,`, `=`, `{`, or `}`, found `:`
   --> $DIR/issue-103869.rs:2:8
    |
+LL | enum VecOrMap{
+   |      -------- while parsing this enum
 LL |     vec: Vec<usize>,
    |        ^ expected one of `(`, `,`, `=`, `{`, or `}`
    |
    = help: enum variants can be `Variant`, `Variant = <integer>`, `Variant(Type, ..., TypeN)` or `Variant { fields: Types }`
```

Comment:
I wonder if this case could be proactively checked for "might be a struct?" in the parser, but not needed in this PR at all.

Reply:
yes, it's a reasonable suggestion, I will do it in a separate PR.